src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryDelegatorFunction.java

```diff
@@ -207,12 +207,12 @@ public SkyValue compute(SkyKey skyKey, Environment env)
     Rule rule;
     try {
       rule = getRepository(repositoryName, env);
-    } catch (ExternalRuleNotFoundException e) {
+      if (rule == null) {
```

Comment:
nit: why move this check here?

Reply:
It felt better to me to have it closer to the actual call, than outside the try/catch block. It's not a huge change.